tests/ert_tests/ert3/data/test_transformations.py

```diff
@@ -36,8 +37,10 @@ def record_factory(is_dir: bool):
             _files = [dir_path / "a.txt", dir_path / "b.txt"]
             with file_factory_context(tmpdir) as file_factory:
                 file_factory(_files)
-            # Using the private sync version for simplicity in tests:
-            tardata = _sync_make_tar(dir_path)
+            tar_obj = io.BytesIO()
```

Comment:
Can we use directly transformation here instead; ie. `TarRecordTransformation.transform_output(mime, dir_path)`

Reply:
Yes, but is that not what we are testing? This factory makes the data to test with, or do I see that wrong? It might be better if that is as independent as possible from the code to be tested.

Reply:
Yes, that's true. Just we need to make sure that the code of the previous `_sync_make_tar` is the same as the one we are creating the tar for test.

Reply:
It depends on what we actually want to test.

If we are testing for tar files, it should not matter how they are made. By making them independently the test will detect if somebody changes the code to use zip files.

If we are testing for any type of archive, which could be for instance zip, then you would want to use the same code to make the test data.